### ISSUE TITLE
Unexpected `include` config behaviour since 0.3.0

### ISSUE THREAD

**Issue by JoostKersjes** [vitest-dev/vitest#725]
### Describe the bug

This is my vitest config from `vite.config.ts`:
```typescript
  test: {
    globals: true,
    environment: "happy-dom",
    include: ["**/__tests__/*.test.ts"],
  },
```
Since 0.3.0 this exact config makes Vitest also includes my `**/__tests__/*.spec.ts` files which I use with Cypress, so Vitest tries to run Cypress tests and fails.

I expect with my `include` config that **only** my pattern is used, and not any patterns from the default configuration.

### Reproduction

https://stackblitz.com/edit/node-aakamc?file=vite.config.js

### System Info

```shell
System:
    OS: Linux 5.11 Ubuntu 20.04.3 LTS (Focal Fossa)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-11370H @ 3.30GHz
    Memory: 1.66 GB / 15.32 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 17.1.0 - ~/.nvm/versions/node/v17.1.0/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v17.1.0/bin/yarn
    npm: 8.1.3 - ~/.nvm/versions/node/v17.1.0/bin/npm
  Browsers:
    Firefox: 96.0
  npmPackages:
    @vitejs/plugin-vue: 2.2.0 => 2.2.0 
    vite: 2.8.0 => 2.8.0 
    vitest: 0.3.0 => 0.3.0
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
> I expect with my include config that **only** my pattern is used, and not any patterns from the default configuration.

This was not the intention. Must be a bug.

**Comment by DerYeger** [comment]
This is caused by arrays being merged.
Wasn't the intention to not merge arrays and export the default (array) values, so users can include them only in addition to their own values if required?

**Comment by sheremet-va** [comment]
> This is caused by arrays being merged. Wasn't the intention to not merge arrays and export the default (array) values, so users can include them only in addition to their own values if required?

Yes, this was the intention. I think `deepMerge` shouldn't merge arrays, this gave us so much headache in the past weeks

**Comment by DerYeger** [comment]
> > This is caused by arrays being merged. Wasn't the intention to not merge arrays and export the default (array) values, so users can include them only in addition to their own values if required?
> 
> Yes, this was the intention. I think `deepMerge` shouldn't merge arrays, this gave us so much headache in the past weeks

I'll create a PR.
